src/Platforms/AbstractPlatform.php

```diff
@@ -1370,6 +1370,11 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff): array
         $sql = [];
 
         foreach ($diff->getDroppedForeignKeys() as $foreignKey) {
+            $constraintName = $foreignKey->getObjectName()?->toString();
+            if ($constraintName === null || $constraintName === '') {
+                continue;
```

Comment:
We don't want to do this. It is better to have invalid DDL generated, so the user sees that something is wrong, rather then get an invalid input silently ignored.